From 4374b673a2b2ebbc7042c062b4f87ab49dd14930 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Thu, 27 Aug 2015 23:14:46 +0000
Subject: [PATCH] BCEL-195 addition of hashCode() to generic/Instruction.java
 breaks Targeters. Never make BranchInstructions compare equal Revert
 short-cut equality check because that would allow a single BI to be shared.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/bcel/trunk@1698244 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/bcel6/generic/InstructionComparator.java       | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/main/java/org/apache/commons/bcel6/generic/InstructionComparator.java b/src/main/java/org/apache/commons/bcel6/generic/InstructionComparator.java
index 936f8d3126..d6d49f5f8d 100644
--- a/src/main/java/org/apache/commons/bcel6/generic/InstructionComparator.java
+++ b/src/main/java/org/apache/commons/bcel6/generic/InstructionComparator.java
@@ -35,13 +35,10 @@ public interface InstructionComparator {
 
         @Override
         public boolean equals( Instruction i1, Instruction i2 ) {
-            if (i1 == i2) {
-                return true; // shortcut for identical objects
-            }
             if (i1.getOpcode() == i2.getOpcode()) {
                 if (i1 instanceof BranchInstruction) {
-                 // Different BIs are never equal to make targeters work correctly (BCEL-195)
-                    return false; // identity checked above
+                 // BIs are never equal to make targeters work correctly (BCEL-195)
+                    return false;
                 } else if (i1 instanceof ConstantPushInstruction) {
                     return ((ConstantPushInstruction) i1).getValue().equals(
                             ((ConstantPushInstruction) i2).getValue());
